apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsList/index.tsx

```diff
@@ -23,16 +24,21 @@ const CommitDetailsList: React.FC<CommitDetailsListProps> = ({
     ) : (
       <>
         {tasks.map((t) => {
-          const { inactiveTasks, task } = t;
+          const { inactiveTasks, shouldShowDateSeparator, task } = t;
           if (task) {
             return (
-              <CommitDetailsCard
-                key={task.id}
-                isCurrentTask={task.id === currentTask.id}
-                owner={currentTask.project?.owner}
-                repo={currentTask.project?.repo}
-                task={task}
-              />
+              <>
+                {shouldShowDateSeparator && (
+                  <DateSeparator date={task.createTime} />
+                )}
+                <CommitDetailsCard
+                  key={task.id}
+                  isCurrentTask={task.id === currentTask.id}
+                  owner={currentTask.project?.owner}
+                  repo={currentTask.project?.repo}
+                  task={task}
+                />
+              </>
             );
           } else if (inactiveTasks) {
```

Comment:
is this block also missing `shouldShowDateSeparator` clause? It is present in `TaskTimeline` and so I would have guessed their code to be parallel

Reply:
I don't think the inactiveCommitsButton existed when I created this pr so I must have missed it 😢